api/src/resources/Transaction/resolver.ts

```diff
@@ -28,9 +33,59 @@ export class ProductResolver {
     return this.transactionRepo.find();
   }
 
+  @Authorized()
+  @Mutation((returns: void) => TransactionPayload)
+  public async startMembershipTransaction(
+    @Ctx() context: IContext,
+    @Arg("membershipType", (type: void) => MembershipTypes)
```

Comment:
I do not know how generically ask the grapqhl API to give me a payment transaction for a `semester-membership` production or `yearly-membership`. It would either be coded in the client as doing a query on all products and then specifically filtering for those keys, or enforced on the server.